src/include/platform/CHIPDeviceConfig.h

```diff
@@ -228,6 +228,15 @@
 #define CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION 0
 #endif
 
+/**
+ * CHIP_DEVICE_CONFIG_DEVICE_INTERACTION_MODEL_REVISION
+ *
+ * A monothonic number identifying the interaction model revision running on the device.
+ */
+#ifndef CHIP_DEVICE_CONFIG_DEVICE_INTERACTION_MODEL_REVISION
+#define CHIP_DEVICE_CONFIG_DEVICE_INTERACTION_MODEL_REVISION 1
```

Comment:
Should this be part of the `CHIPDeviceConfig.h`? I think it usually contains settings which may different across platforms while the IM revision is probably related to the implementation in `src/app`, right?

Reply:
@bzbarsky-apple ?

Reply:
currently I follow the pattern, like software version in ConfigurationManager to do IM revision, it seems we can only put one macro in src/app/InteractionModelRevision.h without integration with ConfigurationManager, which is enough @bzbarsky-apple

Reply:
I see no reason for this to be configurable other than by doing a pull from the SDK repo, so no need for this to be in the overridable device config or configuration manager.

Reply:
ok, would update it now, thanks

Reply:
updated, thanks